packages/expo-doctor/src/checks/DirectPackageInstallCheck.ts

```diff
@@ -0,0 +1,79 @@
+import { DoctorMultiCheck, DoctorMultiCheckItemBase } from './DoctorMultiCheck';
+import { DoctorCheckParams, DoctorCheckResult } from './checks.types';
+
+export type DirectPackageInstallCheckItem = {
+  packageName: string;
+} & DoctorMultiCheckItemBase;
+
+const baseCheckItem = {
+  getMessage: (packageName: string) =>
+    `The package "${packageName}" should not be installed directly in your project. It is a dependency of other Expo packages, which will install it automatically as needed.`,
+  sdkVersionRange: '*',
+};
+
+const shouldBeInstalledGloballyItem = {
+  getMessage: (packageName: string) =>
+    `The package "${packageName}" should not be installed directly in your project. It may conflict with the globally-installed version.`,
+  sdkVersionRange: '*',
+};
+
+export const directPackageInstallCheckItems: DirectPackageInstallCheckItem[] = [
+  {
+    packageName: 'expo-modules-core',
+    ...baseCheckItem,
+  },
+  {
+    packageName: 'expo-modules-autolinking',
+    ...baseCheckItem,
+  },
+  {
+    packageName: 'expo-dev-launcher',
+    ...baseCheckItem,
+  },
+  {
+    packageName: 'expo-dev-menu',
+    ...baseCheckItem,
+  },
+  {
+    packageName: 'npm',
+    ...shouldBeInstalledGloballyItem,
+  },
+  {
+    packageName: 'yarn',
+    ...shouldBeInstalledGloballyItem,
+  },
```

Comment:
It was based on this conversation here: https://exponent-internal.slack.com/archives/CP8BDK9F1/p1682665047400689

At this point, it's probably 97% of a Doctor-meta-check. I did see another case recently where `npm` was in package.json pinned to an old version that broke `npm why`, so a bunch of Doctor checks bailed.

The best I can tell, it's mostly harmless, but also entirely unnecessary, and very occasionally not harmless. I'll add `pnpm` for now, but, based on the above, I could also see restricting it to just `npm`, since that's the only one I've ever seen break anything.